GPU/GPUTracking/TPCClusterFinder/GPUTPCClusterFinder.cxx

```diff
@@ -114,3 +115,34 @@ size_t GPUTPCClusterFinder::getNSteps(size_t items) const
   }
   return c;
 }
+
+void GPUTPCClusterFinder::PrepareMC()
+{
+  if (mRec->IsGPU() || not mRec->GetDeviceProcessingSettings().runMC) {
+    return;
+  }
+
+  assert(mNMaxClusterPerRow > 0);
+
+  clearMCMemory();
+  mPindexMap = new uint[TPCMapMemoryLayout<decltype(*mPindexMap)>::items()];
+  mPlabelsByRow = new GPUTPCClusterMCInterim[GPUCA_ROW_COUNT * mNMaxClusterPerRow];
+  mPlabelHeaderOffset = new uint[GPUCA_ROW_COUNT];
+  mPlabelDataOffset = new uint[GPUCA_ROW_COUNT];
+}
+
+void GPUTPCClusterFinder::clearMCMemory()
+{
+  if (mPindexMap != nullptr) {
```

Comment:
You don't need the `if`s here, delete[] will not do anything if it is a `nullptr`.
Instead, I'd set the ptrs to `nullptr` afterwards, do avoid double deletes, since this is not the destructor but a function.